improver/calibration/reliability_calibration.py

```diff
@@ -615,10 +687,20 @@ def _calculate_reliability_probabilities(self, reliability_table):
             iris.Constraint(table_row_name="sum_of_forecast_probabilities")
         ).data
 
+        valid_bins = np.where(forecast_count >= self.minimum_forecast_count)
+        if valid_bins[0].size < len(reliability_table.coord("probability_bin").points):
+            (
+                observation_count,
+                forecast_probability_sum,
+                forecast_count,
+            ) = self._combine_undersampled_bins(
+                observation_count, forecast_probability_sum, forecast_count
+            )
+
         # In some bins have insufficient counts, return None to avoid applying
         # calibration.
         valid_bins = np.where(forecast_count >= self.minimum_forecast_count)
-        if valid_bins[0].size != forecast_count.size:
+        if valid_bins[0].size != len(reliability_table.coord("probability_bin").points):
```

Comment:
This line will interact with #1315. The desired line after refactoring on top of #1315 should be:
 ```suggestion
         if valid_bins[0].size < len(reliability_table.coord("probability_bin").points) * minimum_bin_fraction:
```